source/common/network/connection_impl.h

```diff
@@ -152,7 +142,14 @@ class ConnectionImpl : public virtual Connection,
   const uint64_t id_;
   std::list<ConnectionCallbacks*> callbacks_;
   std::list<BytesSentCb> bytes_sent_callbacks_;
-  uint32_t state_{InternalState::ReadEnabled};
+  bool read_enabled_{true};
```

Comment:
Should we use a bit field here to save a bit of memory? I don't feel that strongly about it. I can't really recall why I didn't use a bitfield in the first place.

Reply:
I didn't actually verify my assumption, but by my manual counting, I think I shrunk the size of the struct with this change.  The original uint32_t probably had 4 bytes of padding after it (because a pointer was next), so effective size of 8 bytes.  We have 8 booleans here (because I moved a few to pack better), so the class is probably the same size or smaller.  (sizeof(bool) is 1 on x86_64).

The downside to bit-packing is that in-class initialization isn't allowed, so it's not quite as readable.  And I don't think it would make it smaller in this case (although it would if more bools are added).

Reply:
OK SGTM. I don't feel that strongly about it.